PWGHF/D2H/Macros/ML/train_d2h.py

```diff
@@ -0,0 +1,544 @@
+#!/usr/bin/env python3
+
+# Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+# See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+# All rights not expressly granted are reserved.
+
+# This software is distributed under the terms of the GNU General Public
+# License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+
+# In applying this license CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+
+"""
+file: train_d2h.py
+brief: script for the training of ML models to be used in D2H
+note: inspired by EventFiltering/PWGHF/Macros/train_hf_triggers.py and Run2 macros
+usage: python3 train_d2h.py CONFIG
+author: Alexandre Bigot <alexandre.bigot@cern.ch>, Strasbourg University
+author: Mingyu Zhang <mingyu.zang@cern.ch>, Central China Normal University
+"""
+
+import argparse
+import os
+import pickle
+import sys
+
+import matplotlib.pyplot as plt  # pylint: disable=import-error
+import numpy as np  # pylint: disable=import-error
+import pandas as pd  # pylint: disable=import-error
+import xgboost as xgb  # pylint: disable=import-error
+import yaml  # pylint: disable=import-error
+
+# pylint: disable=import-error
+from hipe4ml import plot_utils
+from hipe4ml.model_handler import ModelHandler
+from hipe4ml.tree_handler import TreeHandler
+from hipe4ml_converter.h4ml_converter import H4MLConverter
+from sklearn.model_selection import train_test_split
```

Comment:
Since`hipe4ml` is not installed by default with O2Physics, you could start by trying to import it with something like this
```python
try:
    import hipe4ml
except ModuleNotFoundError:
    print("module 'hipe4ml' is not installed. Please install it to run this macro")
```
The other dependencies should be bundled with `hipe4ml` (to be checked for `yaml`).